Develop3

AlexeyLandik opened the pull request:
Диплом 3. Веб-приложение

AlexeyLandik committed:
Диплом 3. Веб-приложение.

AlexeyLandik committed:
Диплом 3. Веб-приложение.

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,16 @@
+class Urls:
+    BASE_PAGE_URL = 'https://stellarburgers.nomoreparties.site/'
+    FORGOT_PASSWORD_PAGE = 'https://stellarburgers.nomoreparties.site/forgot-password'
+    RESET_PASSWORD_PAGE = 'https://stellarburgers.nomoreparties.site/reset-password'
+    LOGIN_PAGE = 'https://stellarburgers.nomoreparties.site/login'
+    ACCOUNT_PAGE = 'https://stellarburgers.nomoreparties.site/account'
+    ACCOUNT_PROFILE_PAGE = 'https://stellarburgers.nomoreparties.site/account/profile'
+    ORDER_HISTORY_PAGE = 'https://stellarburgers.nomoreparties.site/account/order-history'
+    ORDER_LIST_PAGE = 'https://stellarburgers.nomoreparties.site/feed'
+
+    API_USER_REGISTRATION = 'https://stellarburgers.nomoreparties.site/api/auth/register'
+    API_USER_DELETE = 'https://stellarburgers.nomoreparties.site/api/auth/user'
```
Коммент из 2 проекта

AlexKitIt commented on pages/main_functionality_page.py:
```diff
@@ -0,0 +1,59 @@
+from pages.base_page import BasePage
+import allure
+from seletools.actions import drag_and_drop
+from locators.personal_account_page_locators import PersonalAccountPageLocators
+from locators.main_functionality_page_locators import MainFunctionalityLocators
```
Нужно исправить: здесь и далее: в методах пейджи описываются только методы работы с элементами на конкретной пейдже, работы с элементами с других страниц быть не должно 
Это MainFunctionality а локаторы с PersonalAccountPageLocators

AlexeyLandik committed:
Диплом 3. Веб-приложение. Исправленный

AlexKitIt commented on locators/order_list_page_locators.py:
```diff
@@ -0,0 +1,22 @@
+                            "@class='counter_counter__num__3nue1']")
+    ORDER_BASKET = (By.XPATH, "//ul[@class='BurgerConstructor_basket__list__l9dp_']")
+    BUTTON_ACCOUNT = (By.XPATH, "//*[contains(text(), 'Личный Кабинет')]")
+    BUTTON_ENTER = (By.XPATH, "//button[text()='Войти']")
```
Нужно исправить: здесь и далее: элемент всегда располагается на одной странице, он не копипастится из пейджи в пейджу, а для работы с ним в тесте создается объект конкретной страницы, происходит взаимодействие, потом создается другой объект страницы и взаимодействие с другими элементами. Одинаковые локаторы из разных пейджей необходимо выпилить и оставить только в одном объекте

AlexKitIt commented on pages/order_list_page.py:
```diff
@@ -0,0 +1,137 @@
+from seletools.actions import drag_and_drop
+from data import Urls
+from locators.order_list_page_locators import OrderListLocators
+# from locators.main_functionality_page_locators import MainFunctionalityLocators
+# from locators.personal_account_page_locators import PersonalAccountPageLocators
```
Нужно исправить: закомменченного кода быть не должно

AlexeyLandik committed:
Диплом 3. Веб-приложение. Исправлено:
1. Исключена работа с элементами других пейджей. При необходимости создаются объекты классов других пейджей и идет работа с их методами.
2. Одинаковые локаторы из разных пейджей выпилены.
3. Закомментированный код удален.
